### ISSUE TITLE
Int8/Int32 ReLU

### ISSUE THREAD

**Issue by pranav-prakash** [onnx/onnx#3111]
# New Operator

### Describe the operator
Currently, Relu op only supports float – it would be useful to support integers as well (particularly int8 and int32).

### Can this operator be constructed using existing onnx operators?
It should be straightforward to change the type of Relu to include this. It will probably have to be done in a new opset version though.

### Is this operator used by any model currently? Which one?
Quantized models will often have a QLinearConv followed by Relu and/or Maxpool. While Maxpool supports `int8`, Relu does not, making it impossible to represent these types of models using the standard onnx opset (`uint8` does not suffer from this since the relu can be elided).

### Are you willing to contribute it? (Y/N)
Yes – it should be a straightforward schema change (and corresponding op kernel implementation is trivial). I'm not familiar with the procedure to add a new opset version though.

**Comment by askhade** [comment]
@pranav-prakash : Please send a PR for this so that Operator Sig can review. 
You can follow this PR as an example to understand how to add new opset version : https://github.com/onnx/onnx/pull/2461
Essentially these 2 files need changes : 
onnx/defs/operator_sets.h 
onnx/defs/schema.h

**Comment by pranav-prakash** [comment]
Created PR https://github.com/onnx/onnx/pull/3141 which bumps the opset and updates the schema for Relu.

**Comment by jay-karan** [comment]
Hi @askhade , I have a custom op made from primitive ops using a function. So, do I need to separately register for each datatype? (Say, for Int8 separately, double separately)?

**Comment by askhade** [comment]
@jay-karan : Not sure I fully understand your question. Take a look at this function:
https://github.com/onnx/onnx/blob/master/onnx/defs/quantization/defs.cc#L75

You can check how the type constraints are defined. If this does not help then please paste your function schema here so that I can take a look.

**Comment by jay-karan** [comment]
Thanks @askhade, I have actually followed the similar approach for my operator. I have defined float, float16 and double as Type Constraints in ONNX.

To test the operator as a node. I have registered and implemented the kernel for the it in ONNX Runtime. So, When I was passing float16 or double values as tensors, the operator is stating that "it is expecting only float, but given double/float16". So, do I need to again register separately for each datatype in ONNX Runtime?
